dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserQueryParams.java

```diff
@@ -75,6 +75,11 @@ public class UserQueryParams {
 
   @ToString.Include private UserInvitationStatus invitationStatus;
 
+  /**
+   * WARNING: If this field has an empty collection, it can potentially result in all users being
+   * returned from the database. Check the generated query to make sure you're seeing expected
+   * queries and results.
+   */
```

Comment:
I think this phrasing is a bit misleading. This is one of many potential filters that when empty does not filter away anything. That sounds like expected behaviour to me as long as it is clear that empty means "not set" and not "match against an empty set". I think the comment we leave here should make that clear.

Reply:
I've updated the comment, keeping it broad enough to still help people when debugging. (this would have saved me some time debugging).
If it's still not acceptable, please suggest an alternative.